test/integration/targets/ec2_group/tasks/main.yml

```diff
@@ -325,6 +325,31 @@
           - 'not result.changed'
           - 'result.group_id.startswith("sg-")'
 
+    - name: add a rule that auto creates another security group
```

Comment:
Indeed - as I find issues, I'm doing TDD - add a test, make sure it fails, fix the code, make sure the test passes.